File: server/app/Models/Kegiatan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Kegiatan extends Model
{
    protected $fillable = [
        'judul', 'deskripsi', 'gambar_url', 'tanggal_acara', 'lokasi', 'ukm_id',
    ];

    public function ukm()
    {
        return $this->belongsTo(Ukm::class);
    }
}

File: client/lib/models/kegiatan.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:client/models/ukm.dart';

part 'kegiatan.g.dart';

@JsonSerializable(explicitToJson: true)
class Kegiatan {
  final int id;
  final String judul;
  final String deskripsi;

  @JsonKey(name: 'gambar_url')
  final String? gambarUrl;
  
  @JsonKey(name: 'tanggal_acara')
  final DateTime tanggalAcara;

  final String lokasi;
  
  final Ukm? ukm;

  Kegiatan({
    required this.id,
    required this.judul,
    required this.deskripsi,
    this.gambarUrl,
    required this.tanggalAcara,
    required this.lokasi,
    this.ukm,
  });

  factory Kegiatan.fromJson(Map<String, dynamic> json) => _$KegiatanFromJson(json);

  Map<String, dynamic> toJson() => _$KegiatanToJson(this);
}